nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -0,0 +1,221 @@
+define(['StagingServiceClient', 'common/html', 'common/runtime', 'common/ui', 'util/string'], (
+    StagingServiceClient,
+    html,
+    Runtime,
+    UI,
+    String
+) => {
+    'use strict';
+    const div = html.tag('div'),
+        p = html.tag('p'),
+        ul = html.tag('ul'),
+        li = html.tag('li'),
+        cssBaseClass = 'kb-xsv-gen';
+
+    /**
+     * This widget allows the user to generate a downloadable template suitable for using
+     * with the xSV bulk import interface.
+     *
+     * @param {object} options has keys:
+     *     model: cell metadata
+     * @returns
+     */
+    class XSVGenerator {
+        /**
+         *
+         * @param {object} options configuration options, including
+         *      model           object  cell metadata in one handy Props object
+         */
+        constructor(options) {
+            this._init(options);
+        }
+
+        _init(options) {
+            ['model', 'typesToFiles'].forEach((prop) => {
+                if (!options || !options[prop]) {
+                    throw new Error(`XSV Generator requires the param "${prop}" for instantiation`);
+                }
+                this[prop] = options[prop];
+            });
+            this.runtime = Runtime.make();
+            this.cssBaseClass = cssBaseClass;
+        }
+
+        async run() {
+            const body = this.renderLayout();
+            const confirmed = await UI.showConfirmDialog({
+                title: 'Create CSV Template',
+                body,
+                okLabel: 'Create Template',
+            });
+            // return if the user clicks 'cancel'
+            if (!confirmed) {
+                return;
+            }
```

Comment:
Just added some very bare bones functionality -- ATM this just shows up a dialog box with a message for the user.